.travis.yml

```diff
@@ -9,35 +11,30 @@ python:
 cache: ccache
 compiler: gcc
 
-before_install: # Use this to prepare the system to install prerequisites or dependencies
-  # Define some config vars
-  - export ROS_DISTRO=melodic
-  - export ROS_REPO=ros
-
-  - export NOKOGIRI_USE_SYSTEM_LIBRARIES=true
-  - export REPOSITORY_NAME=${PWD##*/}
-  - echo "Testing branch $TRAVIS_BRANCH of $REPOSITORY_NAME"
-  - sudo -E sh -c 'echo "deb http://packages.ros.org/ros/ubuntu `lsb_release -cs` main" > /etc/apt/sources.list.d/ros-latest.list'
-  - wget http://packages.ros.org/ros.key -O - | sudo apt-key add -
-  - sudo apt-get update -qq
-  - sudo apt-get install -qq -y python-rosdep python-wstool python-catkin-tools
-  # Setup rosdep
-  - sudo rosdep init
-  - rosdep update
-  # Install htmlpoofer
-  - gem update --system
-  - gem --version
-  - gem install html-proofer
-  # Install ROS's version of sphinx
-  - sudo apt-get -qq install ros-kinetic-rosdoc-lite
-  - source /opt/ros/kinetic/setup.bash
+# Define some config vars
+env:
+  global:
+    - MOVEIT_CI_TRAVIS_TIMEOUT=85  # Travis grants us 90 min, but we add a safety margin of 5 min
+    - ROS_REPO=ros
+    - ROS_DISTRO=melodic
+    - UPSTREAM_WORKSPACE=moveit_tutorials.rosinstall
+    - NOKOGIRI_USE_SYSTEM_LIBRARIES=true
+    - CXXFLAGS="-Wall -Wextra -Wwrite-strings -Wunreachable-code -Wpointer-arith -Wredundant-decls -Wno-unused-parameter -Wno-unused-but-set-parameter -Wno-unused-function"
+    - WARNINGS_OK=false
+    - REPOSITORY_NAME=${PWD##*/}
 
+# before_install: # Use this to prepare the system to install prerequisites or dependencies
+  # - ./travis.sh
 before_script:
   - git clone -q --depth=1 https://github.com/ros-planning/moveit_ci.git .moveit_ci
 
 script:
   # Build tutorial examples
   - .moveit_ci/travis.sh
+  # Install htmlpoofer
```

Comment:
The different sections (before_install, before_script, script, etc.) have a particular meaning in Travis.
You shouldn't move instructions around between these sections.

Reply:
Agreed. I tagged this WIP so that I could prototype a fix that passes travis. I didn't expect to get a review yet. :)